samcli/lib/sync/infra_sync_executor.py

```diff
@@ -272,6 +274,27 @@ def _auto_skip_infra_sync(
             LOG.debug("The current template is different from the last deployed version, we will not skip infra sync")
             return False
 
+        try:
+            current_stack_parameters = self._get_stack_parameters(stack_name)
+        except ClientError as ex:
+            LOG.debug("Unable to fetch stack Parameters from stack with name %s", stack_name, exc_info=ex)
+            return False
+
+        # Current stack parameters returned from describe_stacks take the format of e.g.
+        # [{'ParameterKey': 'Foo1', 'ParameterValue': 'Bar1'}, {'ParameterKey': 'Foo2', 'ParameterValue': 'Bar2'}]
+        # Whereas parameter overrides are {'Foo1': 'Bar1', 'Foo2': 'Bar2'}
+        # So we can flatten the current stack parameters into the same format as the parameter overrides
+        # This allows us to check if the parameter overrides are a direct subset of the current stack parameters
+
+        flat_current_stack_parameters = {}
+        for param in current_stack_parameters:
+            flat_current_stack_parameters[param["ParameterKey"]] = param["ParameterValue"]
+
+        # Check for parameter overrides being a subset of the current stack parameters
+        if not (parameter_overrides.items() <= flat_current_stack_parameters.items()):
+            LOG.debug("Detected changes between Parameter overrides and the current stack parameters.")
+            return False
+
```

Comment:
This function (`_auto_skip_infra_sync`) seems a little bit bloated now, do you think moving this logic into its own function would be better?

Reply:
Will do